sdk/tracing/src/main/java/io/opentelemetry/sdk/trace/export/BatchSpanProcessor.java

```diff
@@ -232,10 +232,12 @@ public void run() {
             } while (spansList.isEmpty());
           }
           spansCopy = new ArrayList<>(spansList);
-          spansList.clear();
         }
         // Execute the batch export outside the synchronized to not block all producers.
-        exportBatches(spansCopy);
+        int taken = onBatchExport(spansCopy);
+        synchronized (monitor) {
+          spansList.retainAll(spansList.subList(taken, spansList.size()));
```

Comment:
There should be a better way of doing this. In Scala, I'd resort to using `drop(taken)`.